Doppler.Push.Api/Services/FirebaseCloudMessageService.cs

```diff
@@ -0,0 +1,41 @@
+using Doppler.Push.Api.Contract;
+using FirebaseAdmin.Messaging;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Doppler.Push.Api.Services
+{
+    public class FirebaseCloudMessageService : IFirebaseCloudMessageService
+    {
+        public async Task<MessageSendResponse> SendMulticast(MessageSendRequest request)
+        {
+            var message = new MulticastMessage()
+            {
+                Notification = new Notification()
+                {
+                    Title = request.NotificationTitle,
+                    Body = request.NotificationBody,
+                },
+                Tokens = request.Tokens,
+            };
+
+            var response = await FirebaseMessaging.DefaultInstance.SendMulticastAsync(message);
+
+            var returnResponse = new MessageSendResponse();
+            returnResponse.Responses = response.Responses.Select(x => new ResponseItem
+            {
+                IsSuccess = x.IsSuccess,
+                MessageId = x.MessageId,
+                Exception = x.IsSuccess ? null : new ExceptionItem
+                {
+                    Message = x.Exception.Message,
+                    MessagingErrorCode = (int?)x.Exception.MessagingErrorCode
+                }
+            });
+            returnResponse.FailureCount = response.FailureCount;
+            returnResponse.SuccessCount = response.SuccessCount;
```

Comment:
You can use the initializer:

```suggestion
            var returnResponse = new MessageSendResponse()
            {
                Responses = response.Responses.Select(x => new ResponseItem
                {
                    IsSuccess = x.IsSuccess,
                    MessageId = x.MessageId,
                    Exception = x.IsSuccess ? null : new ExceptionItem
                    {
                        Message = x.Exception.Message,
                        MessagingErrorCode = (int?)x.Exception.MessagingErrorCode
                    }
                }),
                FailureCount = response.FailureCount,
                SuccessCount = response.SuccessCount
            }
```

Reply:
ok, good